addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml

```diff
@@ -552,7 +586,100 @@ Resources:
       VpcEndpointType: Interface
       PrivateDnsEnabled: true
       ServiceName: !Sub 'aws.sagemaker.${AWS::Region}.notebook'
+      SecurityGroupIds:
+        - !Ref SageMakerSecurityGroup
+      VpcId: !Ref VPC
+
+  SageMakerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Condition: isAppStream
+    Properties:
+      GroupDescription: 'SWB SageMaker Security Group for interface endpoint'
+      GroupName: 'SageMakerSG'
+      VpcId: !Ref VPC
+      SecurityGroupIngress:
+        - SourceSecurityGroupId: !Ref AppStreamSecurityGroup
+          IpProtocol: '-1'
+
+  AppStreamSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Condition: isAppStream
+    Properties:
+      GroupDescription: 'SWB AppStream Security Group'
+      GroupName: 'AppStreamSG'
       VpcId: !Ref VPC
+      SecurityGroupEgress:
+        - CidrIp: 127.0.0.1/32
+          IpProtocol: '-1'
+
+  AppStreamSecurityGroupEgress:
+    Type: AWS::EC2::SecurityGroupEgress
+    Condition: isAppStream
+    Properties:
+      GroupId: !Ref AppStreamSecurityGroup
```

Comment:
How does `GroupId` and `DestinationSecurityGroupId` work? I notice we reference different value for these two fields.

Reply:
We are setting egress from `GroupId` to `DestinationSecurityGroupId`. Note that this is a different `Type` than standard SecurityGroup. We are allowing AppStream to connect with SageMaker